src/components/internalSettings/InternalSettings.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from "react";
+import { Accordion, Button, Stack } from "@chakra-ui/react";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+
+import AccordionSection from "./AccordionSection";
+
+const InternalSettings: React.FC = () => {
+  const [branches, setBranches] = useState<any[]>([]);
+  const [branchDict, setBranchDict] = useState<Record<string, any>>({});
+  const { hexathonId } = useParams();
+  const saveBranches = async () => {
+    const promises: any[] = [];
+    for (const branchId of Object.keys(branchDict)) {
+      promises.push(
+        axios.patch(
+          `https://registration.api.hexlabs.org/branches/${branchId}`,
+          branchDict[branchId]
+        )
+      );
+    }
+    await Promise.all(promises);
+  };
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const res = await axios.get(`https://registration.api.hexlabs.org/applications/`);
+        const allBranches: any[] = [];
+        for (const element of res.data) {
+          if (element.hexathon === hexathonId) {
+            const applicationBranchName = element.applicationBranch.name;
+            if (element.confirmationBranch) {
+              const confirmationBranchName = element.confirmationBranch.name;
+              if (!allBranches.includes(confirmationBranchName)) {
+                allBranches.push(element.confirmationBranch);
+              }
+            }
+            if (!allBranches.includes(applicationBranchName)) {
+              allBranches.push(element.applicationBranch);
+            }
+          }
+        }
+        setBranches(allBranches);
+      } catch (e: any) {
+        console.log(e.message);
+      }
+    };
+    getData();
+  }, [hexathonId]);
```

Comment:
Instead of fetching the data like this with `useEffect`, instead use `useAxios`. It'll make it easier and you won't need to have separate state variables. You can look at the other files in the code base for examples.